components/camel-dynamic-router/src/main/java/org/apache/camel/component/dynamicrouter/DynamicRouterConfiguration.java

```diff
@@ -131,6 +129,103 @@ public class DynamicRouterConfiguration {
     @UriParam(label = "common", defaultValue = "false")
     private boolean warnDroppedMessage;
 
+    /**
+     * The ID of the route.
+     */
+    @UriParam(label = "common")
+    private String routeId;
```

Comment:
Why do you need this option and if so, then can you make a better description

Reply:
I suppose it's not needed, although it has been there since the first introduction of the component.  I deleted it now.

Reply:
Camel will automatic inject the route id on processors if they are RouteIdAware - this id should NOT be exposed as a uri parameter for a component.

Reply:
Thanks.  I removed it.